TestApp/MainScreen.js

```diff
@@ -97,10 +97,16 @@ Crashes.setListener({
 
   getErrorAttachments(report) {
     console.log(`Get error attachments for report with id: ${report.id}'`);
-    return [
-      ErrorAttachmentLog.attachmentWithText('hello', 'hello.txt'),
-      ErrorAttachmentLog.attachmentWithBinary(testIcon, 'icon.png', 'image/png')
-    ];
+    return new Promise(async (resolve, reject) => {
```

Comment:
It is an anti-pattern to use async inside Promise. Because ES2017 `async` function returns a Promise and you don't care about `reject` here, this code can be simplified with the following:

```javascript
    return (async () => {
      const textAttachment = await AttachmentsProvider.getTextAttachment();
      const binaryAttachment = await AttachmentsProvider.getBinaryAttachment();
      const binaryName = await AttachmentsProvider.getBinaryName();
      const binaryType = await AttachmentsProvider.getBinaryType();
      return [ErrorAttachmentLog.attachmentWithText(textAttachment, 'hello.txt'),
        ErrorAttachmentLog.attachmentWithBinary(binaryAttachment, binaryName, binaryType)];
    })();
```

OR

```javascript
    const errorAttachmentLogs = async () => {
      const textAttachment = await AttachmentsProvider.getTextAttachment();
      const binaryAttachment = await AttachmentsProvider.getBinaryAttachment();
      const binaryName = await AttachmentsProvider.getBinaryName();
      const binaryType = await AttachmentsProvider.getBinaryType();
      return [ErrorAttachmentLog.attachmentWithText(textAttachment, 'hello.txt'),
        ErrorAttachmentLog.attachmentWithBinary(binaryAttachment, binaryName, binaryType)];
    };
    return errorAttachmentLogs();
```